hazelcast/src/main/java/com/hazelcast/spi/impl/AbstractCompletableFuture.java

```diff
@@ -110,25 +163,71 @@ public V get() throws InterruptedException, ExecutionException {
         }
     }
 
+    @Override
+    public V get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException, TimeoutException {
+        final long deadlineTimeMillis = System.currentTimeMillis() + unit.toMillis(timeout);
+        long millisToWait;
+        for (; ; ) {
+            Object currentState = this.state;
+
+            if (isCancelledState(currentState)) {
+                throw new CancellationException();
+            }
+            if (isDoneState(currentState)) {
+                return getResult();
+            }
+            if (Thread.interrupted()) {
+                throw new InterruptedException();
+            }
+
+            millisToWait = deadlineTimeMillis - System.currentTimeMillis();
+            if (millisToWait <= 0) {
+                throw new TimeoutException();
+            }
+
+            synchronized (this) {
+                if (!isDoneState(this.state)) {
```

Comment:
@jerrinot My suggestion was `isFinalState` to imply that no more state changes can occur, but avoid the implications of having completed successfully. But I must say I also argued for the removal of comments explaining what "done" means for a Future because it is already documented in the official Oracle docs for `Future`. OTOH I do suggest writing a short Javadoc comment on top of `isDoneState` which reinforces the meaning.